Assignment1/src/test/java/files/ResourceTests.java

```diff
@@ -1,118 +1,109 @@
 package files;
-import static io.restassured.RestAssured.*;
 
-import java.util.ArrayList;
-import java.util.List;
+import static io.restassured.RestAssured.given;
 
-import org.apache.http.HttpRequest;
 import org.hamcrest.Matchers;
 import org.junit.Assert;
 import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
-import org.testng.asserts.SoftAssert;
 
-import io.cucumber.core.gherkin.messages.internal.gherkin.internal.com.eclipsesource.json.JsonArray;
-import io.restassured.RestAssured;
 import io.restassured.builder.RequestSpecBuilder;
 import io.restassured.builder.ResponseSpecBuilder;
 import io.restassured.http.ContentType;
-import io.restassured.parsing.Parser;
-import io.restassured.path.json.JsonPath;
-import io.restassured.response.Response;
 import io.restassured.specification.RequestSpecification;
 import io.restassured.specification.ResponseSpecification;
 
 public class ResourceTests {
-	RequestSpecification req = new RequestSpecBuilder().setBaseUri("https://reqres.in/").setContentType(ContentType.JSON).build();
-	ResponseSpecification res200 = new ResponseSpecBuilder().expectStatusCode(200).expectContentType(ContentType.JSON).build();
-	ResponseSpecification res404 = new ResponseSpecBuilder().expectStatusCode(404).expectContentType(ContentType.JSON).build();
-		
-	@Test(dataProvider = "validIds" )
+	RequestSpecification req = new RequestSpecBuilder().setBaseUri("https://reqres.in/")
+			.setContentType(ContentType.JSON).build();
+	ResponseSpecification res200 = new ResponseSpecBuilder().expectStatusCode(200).expectContentType(ContentType.JSON)
+			.build();
+	ResponseSpecification res404 = new ResponseSpecBuilder().expectStatusCode(404).expectContentType(ContentType.JSON)
+			.build();
+
+	@Test(dataProvider = "validIds")
 	private void test_get_singleResource_withValidId_returns_HTTP_200_UStory5(int validId) {
-		// Get request invalid Resource getCall 6			
-		String validResource = given().queryParam("id", validId).spec(req)
-				.when()
-				.get("/api/unknown/")
-				.then().assertThat().statusCode(200).body("isEmpty()", Matchers.is(false))
-				.extract().response().asString();
-	}			
-	
-	@Test (dataProvider= "inValidIds")
+		// Get request invalid Resource getCall 6
+		String validResource = given().queryParam("id", validId).spec(req).when().get("/api/unknown/").then()
```

Comment:
If you are not going to do anything with the variable `validResource` you don't need to define it. In that case your test can end at `...Matchers.is(false))`. The rest of the stuff is not needed.

Reply:
removed unused variable